### ISSUE TITLE
Allow null timeout in pinMessage for unbounded expiration

### ISSUE THREAD

**Issue by ryanheise** [GetStream/stream-chat-flutter#447]
**Is your feature request related to a problem? Please describe.**

The pre-null-safety version used to allow a `null` timeout to effectively make it unbounded.

In the null safety preview this parameter was made non-nullable so you can't indicate `null` anymore.

**Describe the solution you'd like**

Although the parameter is now non-nullable, the internal logic in that method is still the same and has code to handle the `null` case:

```dart
  Future<UpdateMessageResponse> pinMessage(
    Message message,
    Object timeoutOrExpirationDate,
  ) {
    assert(() {
      if (timeoutOrExpirationDate is! DateTime &&
          timeoutOrExpirationDate is! num) {
        throw ArgumentError('Invalid timeout or Expiration date');
      }
      return true;
    }(), 'Check whether time out is valid');

    DateTime? pinExpires;
    if (timeoutOrExpirationDate is DateTime) {
      pinExpires = timeoutOrExpirationDate.toUtc();
    } else if (timeoutOrExpirationDate is num) {
      pinExpires = DateTime.now()
          .add(
            Duration(seconds: timeoutOrExpirationDate.toInt()),
          )
          .toUtc();
    }
    return updateMessage(
      message.copyWith(pinned: true, pinExpires: pinExpires),
    );
  }
```

So even though `null` is not a possible parameter, anything that is not a `DateTime` or a `num` will be treated like the old `null` behaviour. So we could for example pass in an empty string `''` instead of `null` and we can trigger that internal logic to allow an unbounded expiration.

So it would be nice to just fix this by making the parameter nullable and removing the null assertion check to allow it.

**Describe alternatives you've considered**

I can specify a really large expiration date, but I think the old behaviour was more elegant.

One more alternative "proposal" would be to make this second parameter optional, defaulting to null. This would be a backwards compatible change.

**Additional context**

None.

**Comment by imtoori** [comment]
Thanks for reporting this bug.
It will be fixed in the next release

**Comment by xsahil03x** [comment]
Hey @ryanheise, thanks for reporting we might have missed this while migrating to NNBD. Meanwhile, you can use `channel.pinMessage` instead of `client.pinMessage` if it's urgent for you.

**Comment by xsahil03x** [comment]
Closing as fixed in #448
